torch/distributed/checkpoint/state_dict.py

```diff
@@ -577,7 +578,7 @@ def _load_model_state_dict(
         return cast(
             _IncompatibleKeys,
             _state_dict_fn(model, "load_state_dict")(
-                state_dict=state_dict, strict=info.strict
+                state_dict=state_dict, strict=info.strict, assign=assign
```

Comment:
`assign = True` could avoid the memory cost from `model.to_empty(device=device)` for meta device